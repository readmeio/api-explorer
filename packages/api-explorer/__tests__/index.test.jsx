const React = require('react');
const { shallow, mount } = require('enzyme');
const Cookie = require('js-cookie');
const extensions = require('@readme/oas-extensions');
const WrappedApiExplorer = require('../src');

const { ApiExplorer } = WrappedApiExplorer;

const oas = require('./fixtures/petstore/oas');
const oasCommon = require('./fixtures/parameters/common');

const createDocs = require('../lib/create-docs');

const docs = createDocs(oas, 'api-setting');

const languages = ['node', 'curl'];
const props = {
  appearance: {},
  docs,
  flags: {},
  glossaryTerms: [],
  oasFiles: {
    'api-setting': { ...oas, [extensions.SAMPLES_LANGUAGES]: languages },
  },
  suggestedEdits: false,
  variables: { user: {}, defaults: [] },
};

test('ApiExplorer renders a doc for each', () => {
  const explorer = shallow(<ApiExplorer {...props} />);

  expect(explorer.find('Doc')).toHaveLength(docs.length);
});

test('ApiExplorer should not render a common parameter OAS operation method', () => {
  const docsCommon = createDocs(oasCommon, 'api-setting');
  const propsCommon = {
    ...props,
    docs: docsCommon,
    oasFiles: {
      'api-setting': oasCommon,
    },
  };

  const explorer = shallow(<ApiExplorer {...propsCommon} />);

  // Doc should have neither `servers` or `parameters` from the spec because those aren't real HTTP methods.
  expect(explorer.find('Doc')).toHaveLength(docsCommon.length - 2);
});

test('should display an error message if it fails to render (wrapped in ErrorBoundary)', () => {
  // Prompting an error with an array of nulls instead of Docs
  // This is to simulate some unknown error state during initial render
  const explorer = mount(<WrappedApiExplorer {...props} docs={[null, null]} />);

  expect(explorer.find('ErrorBoundary')).toHaveLength(1);
});

describe('selected language', () => {
  it('should default to curl', () => {
    const explorer = shallow(
      <ApiExplorer
        {...props}
        oasFiles={{
          'api-setting': oas,
        }}
      />,
    );

    expect(explorer.state('language')).toBe('curl');
  });

  it('should auto-select to the first language of the first oas file', () => {
    const explorer = shallow(<ApiExplorer {...props} />);

    expect(explorer.state('language')).toBe(languages[0]);
  });

  describe('#setLanguage()', () => {
    it('should update the language state', () => {
      const explorer = shallow(<ApiExplorer {...props} />);

      explorer.instance().setLanguage('language');
      expect(explorer.state('language')).toBe('language');
      expect(Cookie.get('readme_language')).toBe('language');
    });
  });

  describe('Cookie', () => {
    it('the state of language should be set to Cookie value if provided', () => {
      Cookie.set('readme_language', 'javascript');
      const explorer = shallow(<ApiExplorer {...props} />);

      expect(explorer.state('language')).toBe('javascript');
    });
  });

  it('the state of language should be the first language defined if cookie has not been set', () => {
    Cookie.remove('readme_language');
    const explorer = shallow(<ApiExplorer {...props} />);

    expect(explorer.state('language')).toBe('node');
  });

  it('the state of language should be defaulted to curl if no cookie is present and languages have not been defined', () => {
    Cookie.remove('readme_language');
    const explorer = shallow(
      <ApiExplorer
        {...props}
        oasFiles={{
          'api-setting': oas,
        }}
      />,
    );

    expect(explorer.state('language')).toBe('curl');
  });
});

describe('oas', () => {
  const baseDoc = {
    _id: 1,
    title: 'title',
    slug: 'slug',
    type: 'endpoint',
    category: {},
    api: { method: 'get' },
  };

  // Swagger apis and some legacies
  it('should fetch it from `doc.category.apiSetting`', () => {
    const explorer = shallow(
      <ApiExplorer
        {...props}
        docs={[{ ...baseDoc, category: { apiSetting: 'api-setting' } }]}
        oasFiles={{
          'api-setting': oas,
        }}
      />,
    );

    expect(explorer.find('Doc').get(0).props.oas).toBe(oas);
  });

  // Some other legacy APIs where Endpoints are created in arbitrary categories
  it('should fetch it from `doc.api.apiSetting._id`', () => {
    const explorer = shallow(
      <ApiExplorer
        {...props}
        docs={[{ ...baseDoc, api: { method: 'get', apiSetting: { _id: 'api-setting' } } }]}
        oasFiles={{
          'api-setting': oas,
        }}
      />,
    );

    expect(explorer.find('Doc').get(0).props.oas).toBe(oas);
  });

  it('should fetch it from `doc.api.apiSetting` if it is a string', () => {
    const explorer = shallow(
      <ApiExplorer
        {...props}
        docs={[{ ...baseDoc, api: { method: 'get', apiSetting: 'api-setting' } }]}
        oasFiles={{
          'api-setting': oas,
        }}
      />,
    );

    expect(explorer.find('Doc').get(0).props.oas).toBe(oas);
  });

  // Of course... `typeof null === 'object'`
  it('should not error if `doc.api.apiSetting` is null', () => {
    const explorer = shallow(
      <ApiExplorer {...props} docs={[{ ...baseDoc, api: { method: 'get', apiSetting: null } }]} />,
    );

    expect(explorer.find('Doc').get(0).props.oas).toStrictEqual({});
  });

  it('should set it to empty object', () => {
    const explorer = shallow(<ApiExplorer {...props} docs={[baseDoc]} />);

    expect(explorer.find('Doc').get(0).props.oas).toStrictEqual({});
  });
});

describe('auth', () => {
  it('should read apiKey from `variables.user.apiKey`', () => {
    const apiKey = '123456';

    const explorer = shallow(<ApiExplorer {...props} variables={{ user: { apiKey } }} />);

    expect(explorer.state('auth')).toStrictEqual({ api_key: '123456', petstore_auth: '123456' });
  });

  it('should read apiKey from `variables.user.keys[].apiKey`', () => {
    const apiKey = '123456';

    const explorer = shallow(
      <ApiExplorer {...props} variables={{ user: { keys: [{ name: 'a', apiKey }] } }} />,
    );

    expect(explorer.state('auth')).toStrictEqual({ api_key: '123456', petstore_auth: '123456' });
  });

  it('should read apiKey from `user_data.keys[].api_key`', () => {
    const apiKey = '123456';

    const explorer = shallow(
      <ApiExplorer
        {...props}
        variables={{ user: { keys: [{ name: 'project1', api_key: apiKey }] } }}
      />,
    );

    expect(explorer.state('auth')).toStrictEqual({ api_key: '123456', petstore_auth: '' });
  });

  it('should return nothing for lazy + `splitReferenceDocs`', () => {
    const explorer = shallow(<ApiExplorer {...props} appearance={{ splitReferenceDocs: true }} />);

    const { lazyHash } = explorer.instance();
    expect(lazyHash).toStrictEqual({});
  });

  it('should disable lazy render first 5 projects', () => {
    const explorer = shallow(<ApiExplorer {...props} />);

    const { lazyHash } = explorer.instance();
    expect(lazyHash[0]).toBe(false);
    expect(lazyHash[1]).toBe(false);
    expect(lazyHash[2]).toBe(false);
    expect(lazyHash[3]).toBe(false);
    expect(lazyHash[4]).toBe(false);
    expect(lazyHash[5]).toBe(true);
  });

  it('should disable lazy render middle 5 projects', () => {
    // somewhere in the middle of props.docs
    window.history.pushState({}, '', '/#user-createWithArray');
    const explorer = shallow(<ApiExplorer {...props} />);

    const instance = explorer.instance();
    const slugs = instance.props.docs.map(x => x.slug);
    const centerIdx = slugs.indexOf('user-createWithArray');

    expect(instance.lazyHash[centerIdx - 3]).toBe(true);
    expect(instance.lazyHash[centerIdx - 2]).toBe(false);
    expect(instance.lazyHash[centerIdx - 1]).toBe(false);
    expect(instance.lazyHash[centerIdx]).toBe(false);
    expect(instance.lazyHash[centerIdx + 1]).toBe(false);
    expect(instance.lazyHash[centerIdx + 2]).toBe(false);
    expect(instance.lazyHash[centerIdx + 3]).toBe(true);
  });

  it('should disable lazy render for last 5 projects', () => {
    // last doc in props.docs
    window.history.pushState({}, '', '/#user-username');
    const explorer = shallow(<ApiExplorer {...props} />);

    const instance = explorer.instance();
    const slugs = instance.props.docs.map(x => x.slug);
    const centerIdx = slugs.indexOf('user-username');

    expect(instance.lazyHash[centerIdx - 3]).toBe(true);
    expect(instance.lazyHash[centerIdx - 2]).toBe(false);
    expect(instance.lazyHash[centerIdx - 1]).toBe(false);
    expect(instance.lazyHash[centerIdx]).toBe(false);
    expect(instance.lazyHash[centerIdx + 1]).toBe(false);
    expect(instance.lazyHash[centerIdx + 2]).toBe(false);
  });

  it('should default to empty string', () => {
    const explorer = shallow(<ApiExplorer {...props} />);

    expect(explorer.state('auth')).toStrictEqual({ api_key: '', petstore_auth: '' });
  });

  it('should be updated via editing authbox', () => {
    const explorer = mount(<ApiExplorer {...props} docs={docs.slice(0, 1)} />);
    const doc = explorer
      .find('Doc')
      .at(0)
      .instance();

    doc.setState({ showEndpoint: true, showAuthBox: true });

    explorer.update();

    const input = explorer.find('input[name="apiKey"]');

    input.instance().value = '1234';
    input.simulate('change');

    expect(explorer.state('auth').petstore_auth).toBe('1234');

    input.instance().value += '5678';
    input.simulate('change');

    expect(explorer.state('auth').petstore_auth).toBe('12345678');
  });

  it('should merge securities auth changes', () => {
    const explorer = mount(<ApiExplorer {...props} />);

    explorer.instance().onAuthChange({ api_key: '7890' });
    explorer.instance().onAuthChange({ petstore_auth: '123456' });

    expect(explorer.state('auth')).toStrictEqual({ api_key: '7890', petstore_auth: '123456' });
  });
});
