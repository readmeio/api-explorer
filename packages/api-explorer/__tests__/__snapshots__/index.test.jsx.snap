// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocAsync should do OAS dereferencing 1`] = `
"<div class=\\"is-lang-curl\\"><div class=\\"content-body hub-reference-sticky hub-reference-theme-undefined\\" id=\\"hub-reference\\"><div class=\\"hub-reference\\" id=\\"page-pets\\"><a class=\\"anchor-page-title\\" id=\\"pets\\"></a><div class=\\"hub-reference-section hub-reference-section-top\\"><div class=\\"hub-reference-left\\"><header><h2>oneOf request with a nested allOf</h2></header></div><div class=\\"hub-reference-right\\">&nbsp;</div></div><div class=\\"hub-api\\"><div class=\\"api-definition-parent\\"><div class=\\"api-definition\\"><div class=\\"api-definition-container\\"><div class=\\"api-definition-actions\\"><button class=\\"api-try-it-out active\\" type=\\"submit\\"><span class=\\"try-it-now-btn\\"><span class=\\"fa fa-compass\\"></span>&nbsp;<span>Try It</span></span></button></div><span class=\\"pg-type-big pg-type type-patch\\">patch</span><span class=\\"definition-url\\"><span class=\\"url\\">https://httpbin.org</span><span class=\\"api-text\\">/pets</span></span></div></div></div><div class=\\"hub-reference-section hub-reference-section-code\\"><div class=\\"hub-reference-left\\"><div class=\\"code-sample tabber-parent\\"><div><ul class=\\"code-sample-tabs\\"><li><a class=\\"hub-lang-switch-curl\\" href=\\"#\\">cURL</a></li><li><a class=\\"hub-lang-switch-node\\" href=\\"#\\">Node</a></li><li><a class=\\"hub-lang-switch-ruby\\" href=\\"#\\">Ruby</a></li><li><a class=\\"hub-lang-switch-javascript\\" href=\\"#\\">JavaScript</a></li><li><a class=\\"hub-lang-switch-python\\" href=\\"#\\">Python</a></li></ul><div class=\\"hub-code-auto\\"><button class=\\"copy-code-button\\" type=\\"button\\"><span class=\\"fa fa-clipboard\\"></span></button><pre class=\\"tomorrow-night hub-lang hub-lang-curl\\"><div class=\\"cm-s-tomorrow-night\\"><span class=\\"cm-builtin\\">curl</span> <span class=\\"cm-attribute\\">--request</span> PATCH \\\\
  <span class=\\"cm-attribute\\">--url</span> https://httpbin.org/pets \\\\
  <span class=\\"cm-attribute\\">--header</span> <span class=\\"cm-string\\">'Content-Type: application/json'</span></div></pre></div></div></div></div><div class=\\"hub-reference-right hub-reference-results tabber-parent\\"><div class=\\"hub-reference-results-slider\\"><div class=\\"hub-reference-results-explorer code-sample\\"></div><div class=\\"hub-reference-results-examples code-sample\\"><div class=\\"hub-no-code\\">Try the API to see Results</div></div></div></div></div><div class=\\"hub-reference-section\\"><div class=\\"hub-reference-left\\"><div class=\\"param-type \\" id=\\"form-patchpets\\"><div class=\\"param-type-header\\"><h3>Body Params</h3><div class=\\"param-header-border\\"></div></div><form class=\\"rjsf\\" id=\\"form-body-patchpets\\"><div class=\\"form-group field field-undefined param\\"><div class=\\"panel panel-default panel-body\\"><div class=\\"form-group\\"><select class=\\"form-control\\" id=\\"body-patchpets__oneof_select\\"><option value=\\"0\\">Option 1</option><option value=\\"1\\">Option 2</option></select></div><div class=\\"form-group field field-object param\\"><fieldset id=\\"body-patchpets\\"><div class=\\"form-group field field-string param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-patchpets_pet_type\\">pet_type<span class=\\"label-required\\">*</span></label><span class=\\"label-type\\">string</span><div class=\\"description\\"></div></span><div class=\\"children\\"><input class=\\"form-control\\" id=\\"body-patchpets_pet_type\\" label=\\"pet_type\\" placeholder=\\"\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div><div class=\\"form-group field field-boolean param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-patchpets_hunts\\">hunts</label><span class=\\"label-type\\">boolean</span><div class=\\"description\\"></div></span><div class=\\"children\\"><select class=\\"form-control\\" id=\\"body-patchpets_hunts\\"><option value=\\"\\"></option><option value=\\"true\\">true</option><option value=\\"false\\">false</option></select></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div><div class=\\"form-group field field-integer  field-int32 param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-patchpets_age\\">age</label><span class=\\"label-type\\">int32</span><div class=\\"description\\"></div></span><div class=\\"children\\"><input class=\\"form-control\\" type=\\"number\\" id=\\"body-patchpets_age\\" label=\\"age\\" placeholder=\\"\\" value=\\"\\"></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div></fieldset><div class=\\"errors\\"></div><div class=\\"help\\"></div></div></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div><button style=\\"display: none;\\" type=\\"submit\\"></button></form></div></div><div class=\\"hub-reference-right switcher\\"><div class=\\"hub-reference-response-definitions\\"><h3><div class=\\"pull-right\\"><select class=\\"switcher-switch\\"><option value=\\"200\\">200</option></select></div>Response</h3><div class=\\"response-schema\\"><div class=\\"desc\\"><p>Updated</p></div></div></div></div></div></div><div class=\\"hub-reference-section\\"><div class=\\"hub-reference-left\\"><div class=\\"content-body\\"><div></div></div></div><div class=\\"hub-reference-right\\"><div class=\\"content-body\\"><div></div></div></div></div><input id=\\"swagger-x-send-defaults\\" type=\\"hidden\\" value=\\"\\"></div><div class=\\"hub-reference\\" id=\\"page-pets\\"><a class=\\"anchor-page-title\\" id=\\"pets\\"></a><div class=\\"hub-reference-section hub-reference-section-top\\"><div class=\\"hub-reference-left\\"><header><h2>allOf with a readOnly prop</h2></header></div><div class=\\"hub-reference-right\\">&nbsp;</div></div><div class=\\"hub-api\\"><div class=\\"api-definition-parent\\"><div class=\\"api-definition\\"><div class=\\"api-definition-container\\"><div class=\\"api-definition-actions\\"><button class=\\"api-try-it-out active\\" type=\\"submit\\"><span class=\\"try-it-now-btn\\"><span class=\\"fa fa-compass\\"></span>&nbsp;<span>Try It</span></span></button></div><span class=\\"pg-type-big pg-type type-put\\">put</span><span class=\\"definition-url\\"><span class=\\"url\\">https://httpbin.org</span><span class=\\"api-text\\">/pets</span></span></div></div></div><div class=\\"hub-reference-section hub-reference-section-code\\"><div class=\\"hub-reference-left\\"><div class=\\"code-sample tabber-parent\\"><div><ul class=\\"code-sample-tabs\\"><li><a class=\\"hub-lang-switch-curl\\" href=\\"#\\">cURL</a></li><li><a class=\\"hub-lang-switch-node\\" href=\\"#\\">Node</a></li><li><a class=\\"hub-lang-switch-ruby\\" href=\\"#\\">Ruby</a></li><li><a class=\\"hub-lang-switch-javascript\\" href=\\"#\\">JavaScript</a></li><li><a class=\\"hub-lang-switch-python\\" href=\\"#\\">Python</a></li></ul><div class=\\"hub-code-auto\\"><button class=\\"copy-code-button\\" type=\\"button\\"><span class=\\"fa fa-clipboard\\"></span></button><pre class=\\"tomorrow-night hub-lang hub-lang-curl\\"><div class=\\"cm-s-tomorrow-night\\"><span class=\\"cm-builtin\\">curl</span> <span class=\\"cm-attribute\\">--request</span> PUT \\\\
  <span class=\\"cm-attribute\\">--url</span> https://httpbin.org/pets \\\\
  <span class=\\"cm-attribute\\">--header</span> <span class=\\"cm-string\\">'Content-Type: application/json'</span></div></pre></div></div></div></div><div class=\\"hub-reference-right hub-reference-results tabber-parent\\"><div class=\\"hub-reference-results-slider\\"><div class=\\"hub-reference-results-explorer code-sample\\"></div><div class=\\"hub-reference-results-examples code-sample\\"><div class=\\"hub-no-code\\">Try the API to see Results</div></div></div></div></div><div class=\\"hub-reference-section\\"><div class=\\"hub-reference-left\\"><div class=\\"param-type \\" id=\\"form-putpets\\"><div class=\\"param-type-header\\"><h3>Body Params</h3><div class=\\"param-header-border\\"></div></div><form class=\\"rjsf\\" id=\\"form-body-putpets\\"><div class=\\"form-group field field-object\\"><fieldset id=\\"body-putpets\\"><div class=\\"form-group field field-object param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-putpets__self\\">_self</label><span class=\\"label-type\\">object</span><div class=\\"description\\"></div></span><div class=\\"children\\"><fieldset id=\\"body-putpets__self\\"><legend id=\\"body-putpets__self__title\\">_self</legend><div class=\\"form-group field field-string param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-putpets__self_href\\">href<span class=\\"label-required\\">*</span></label><span class=\\"label-type\\">string</span><div class=\\"description\\"></div></span><div class=\\"children\\"><input class=\\"form-control\\" readonly=\\"\\" id=\\"body-putpets__self_href\\" label=\\"href\\" placeholder=\\"\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div><div class=\\"form-group field field-string param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-putpets__self_rel\\">rel<span class=\\"label-required\\">*</span></label><span class=\\"label-type\\">string</span><div class=\\"description\\"></div></span><div class=\\"children\\"><input class=\\"form-control\\" readonly=\\"\\" id=\\"body-putpets__self_rel\\" label=\\"rel\\" placeholder=\\"\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div><div class=\\"form-group field field-string param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-putpets__self_title\\">title</label><span class=\\"label-type\\">string</span><div class=\\"description\\"></div></span><div class=\\"children\\"><input id=\\"body-putpets__self_title\\" type=\\"hidden\\" value=\\"\\"></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div><div class=\\"form-group field field-object param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-putpets__self_source\\">source</label><span class=\\"label-type\\">object</span><div class=\\"description\\"></div></span><div class=\\"children\\"><fieldset id=\\"body-putpets__self_source\\"><legend id=\\"body-putpets__self_source__title\\">source</legend><div class=\\"form-group field field-string param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-putpets__self_source_href\\">href<span class=\\"label-required\\">*</span></label><span class=\\"label-type\\">string</span><div class=\\"description\\"></div></span><div class=\\"children\\"><input class=\\"form-control\\" readonly=\\"\\" id=\\"body-putpets__self_source_href\\" label=\\"href\\" placeholder=\\"\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div><div class=\\"form-group field field-string param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-putpets__self_source_rel\\">rel<span class=\\"label-required\\">*</span></label><span class=\\"label-type\\">string</span><div class=\\"description\\"></div></span><div class=\\"children\\"><input class=\\"form-control\\" readonly=\\"\\" id=\\"body-putpets__self_source_rel\\" label=\\"rel\\" placeholder=\\"\\" required=\\"\\" type=\\"text\\" value=\\"\\"></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div><div class=\\"form-group field field-string param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-putpets__self_source_title\\">title</label><span class=\\"label-type\\">string</span><div class=\\"description\\"></div></span><div class=\\"children\\"><input id=\\"body-putpets__self_source_title\\" type=\\"hidden\\" value=\\"\\"></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div></fieldset></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div></fieldset></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div><div class=\\"form-group field field-integer  field-int32 param\\"><span class=\\"label\\"><label class=\\"label-name\\" for=\\"body-putpets_id\\">id</label><span class=\\"label-type\\">int32</span><div class=\\"description\\"></div></span><div class=\\"children\\"><input class=\\"form-control\\" type=\\"number\\" id=\\"body-putpets_id\\" label=\\"id\\" placeholder=\\"\\" value=\\"\\"></div><div class=\\"errors\\"></div><div class=\\"help\\"></div></div></fieldset></div><button style=\\"display: none;\\" type=\\"submit\\"></button></form></div></div><div class=\\"hub-reference-right switcher\\"><div class=\\"hub-reference-response-definitions\\"><h3><div class=\\"pull-right\\"><select class=\\"switcher-switch\\"><option value=\\"200\\">200</option></select></div>Response</h3><div class=\\"response-schema\\"><div class=\\"desc\\"><p>successful operation</p></div></div></div></div></div></div><div class=\\"hub-reference-section\\"><div class=\\"hub-reference-left\\"><div class=\\"content-body\\"><div></div></div></div><div class=\\"hub-reference-right\\"><div class=\\"content-body\\"><div></div></div></div></div><input id=\\"swagger-x-send-defaults\\" type=\\"hidden\\" value=\\"\\"></div></div></div>"
`;
